garage/envs/point_env.py

```diff
@@ -28,13 +33,21 @@ def reset(self):
     def step(self, action):
         self._state = self._state + action
         x, y = self._state
-        reward = -(x**2 + y**2)**0.5
+        reward = -((x - self.goal[0])**2 + (y - self.goal[1])**2)**0.5
         done = abs(x) < 0.01 and abs(y) < 0.01
         next_observation = np.copy(self._state)
         return Step(observation=next_observation, reward=reward, done=done)
 
-    def render(self):
+    def render(self, mode=None):
         print('current state:', self._state)
 
     def log_diagnostics(self, paths):
         pass
+
+    def reset_task(self, task):
```

Comment:
## Don't create interfaces without specifying them
Violating the `gym.Env` interface for convenience is not the way to go here. You have added a new interface (`reset_task`, `task_space`) but did not endorsed anywhere what that interface is.

## Meet existing code where it is -- changes are expensive
It's also very inflexible to force the base `gym.Env` to implement a multitask env interface. The entire purpose of a `MultitaskEnv` *wrapper* should be to *avoid* forcing a new interface onto an existing environment, since most environments are not multitask. You will of course need to assert that all of those envs have the same action space and observation space.

## Separate the API contract from the implementation
Of course, there's nothing wrong with implementing an environment which implements a multitask API directly (it's strictly better), but in the beginning most environments simply won't do that.

If you want an API contract for Multitask environments, you need to say so. You've spread the API across two layers here, in a pretty confusing way. Why is there even a multitask wrapper if you forced PointEnv to implement the multitask interface???

Here's a blueprint for this
* An ABC specifying the set of functions a MultitaskEnv needs to provide (it does not provide any of them itself). 
* A wrapper which takes a list of `gym.Env`'s and turns them into a `MultitaskEnv` (it implements your MultitaskEnv interface)
* (optional) An example of an environment implementing MultitaskEnv directly (MultitaskPointEnv would be a decent option)

My existing implementation in